options/posix/generic/unistd-stubs.cpp

```diff
@@ -292,7 +292,20 @@ int ftruncate(int fd, off_t size) {
 }
 
 char *getcwd(char *buffer, size_t size) {
-	__ensure(buffer && "glibc extension for !buffer is not implemented");
+	/* In order to support glibc's extension of allocating buffer if none is given, we have this
```

Comment:
Maybe rename the commit for these changes, as ``getcwd`` was implemented before. Maybe we can also add a test that compares the output of ``getcwd`` with and without the buffer?

Reply:
Addressed in 86fade5.